providers.h

```diff
@@ -99,8 +102,8 @@ class CommandProvider: public QObject, public QList<CommandProviderItem*>
     virtual bool isOutDated() const { return false; }
 
 signals:
-    void aboutToBeChanged();
-    void changed();
+    void aboutToBeChanged(const QList<QPersistentModelIndex> &parents = QList<QPersistentModelIndex>(), QAbstractItemModel::LayoutChangeHint hint = QAbstractItemModel::NoLayoutChangeHint);
+    void changed(const QList<QPersistentModelIndex> &parents = QList<QPersistentModelIndex>(), QAbstractItemModel::LayoutChangeHint hint = QAbstractItemModel::NoLayoutChangeHint);
```

Comment:
IMHO, this damages code readability. I know why you did it but I think there should be another way, without this change.

Reply:
I'm not sure I understood your idea. I can add two 2 slots (without parameters) in the CommandSourceItemModel class and emit there the 2 signals. I'm not seeing the gain, anyway.

Reply:
> I'm not seeing the gain, anyway.

The gain is a well-written code: the class `CommandProvider` doesn't need to know how its signals will be used; its job is emitting them appropriately.

> I can add two 2 slots

Now that you're changing the connections to Qt5 signal/slot syntax, please use lambda instead. As in:
```c++
connect(provider, &CommandProvider::changed, this, [this] { emit layoutChanged(); });
```

Reply:
Oh... stupid me. That's super easy! And more readable... You rock :1st_place_medal: